simulariumio/writers/binary_writer.py

```diff
@@ -0,0 +1,417 @@
+        current_chunk = 0
+        for frame_index, buffer_size in enumerate(frame_buffer_sizes):
+            frame_values = BINARY_SETTINGS.FRAME_HEADER_LENGTH + buffer_size
+            frame_bytes = BINARY_SETTINGS.BYTES_PER_VALUE * frame_values
+            if frame_bytes > max_bytes:
+                raise Exception(
+                    f"Frame {frame_index} is too large for a simularium file "
+                    f"({frame_bytes} bytes), try filtering out some data."
+                )
+            new_bytes = chunks[current_chunk].n_bytes - n_constant_bytes + frame_bytes
+            if new_bytes > max_bytes:
+                current_chunk += 1
+                chunks.append(
+                    BinaryChunk(0, n_constant_bytes, n_constant_values, frame_index)
+                )
+            chunks[current_chunk].n_frames += 1
+            chunks[current_chunk].n_bytes += frame_bytes
+            chunks[current_chunk].n_values += frame_values
+        for chunk in chunks:
+            data_index = 0
+            for frame_index in range(chunk.n_frames):
+                chunk.frame_offsets.append(data_index)
+                data_index += (
+                    BINARY_SETTINGS.FRAME_HEADER_LENGTH
+                    + frame_buffer_sizes[chunk.get_global_index(frame_index)]
+                )
+        return chunks
+
+    @staticmethod
+    def _get_binary_header(
+        chunk: BinaryChunk,
+        header_length: int,
+        traj_info_length: int,
+        spatial_data_length: int,
+        plot_data_length: int,
+    ) -> BinaryValues:
+        """
+        Return the binary header values and format
+        """
+        header_format = (
+            f"{len(BINARY_SETTINGS.HEADER)}s{BINARY_SETTINGS.N_HEADER_INT_VALUES()}i"
+        )
+        block_offsets = [
+            0,
+            traj_info_length,
+            traj_info_length + spatial_data_length,
+        ]
+        block_offsets = [header_length + offset for offset in block_offsets]
+        return BinaryValues(
+            values=(
+                [bytes(BINARY_SETTINGS.HEADER, "utf-8")]
+                + [header_length, BINARY_SETTINGS.VERSION, BINARY_SETTINGS.N_BLOCKS]
+                + block_offsets
+                + BINARY_SETTINGS.DEFAULT_BLOCK_TYPES()
+                + [traj_info_length, spatial_data_length, plot_data_length]
+            ),
+            format_string=header_format,
+        )
+
+    @staticmethod
+    def _get_binary_spatial_data_header(
+        chunk: BinaryChunk,
+        constant_first_frame_offset: int,
+        spatial_header_constant_size: int,
+        spatial_data_length: int,
+    ) -> BinaryValues:
+        """
+        Return spatial data header values and format
+        """
+        first_frame_offset = constant_first_frame_offset + chunk.n_frames
+        chunk.frame_offsets = [
+            offset + first_frame_offset for offset in chunk.frame_offsets
```

Comment:
@toloudis should the frame offsets be the global offset from the start of the file, or the offset within the spatial data block? I assumed global would be easier for parsing, but wanted to check

Reply:
If this is what I think it is,  then I think it should be offsets within the spatial data block.  Mainly so that we aren't changing too many things. We already have code that expects these offsets to be that.  I'll be able to answer more definitively when I really dig into this code in a couple days.

Reply:
ok let me know and I can make that change

Reply:
yes please